src/features/course-main/course-main.tsx

```diff
@@ -22,27 +23,35 @@ export const CourseMain = ({ courseName, type }: CourseMainProps) => {
     return <p>Error fetching course. Try again.</p>;
   }
 
-  const now = new Date().setHours(0, 0, 0, 0);
-  const requiredDate = new Date(course.startDate).setHours(0, 0, 0, 0);
+  const { title, altTitle, language, mode, enroll, secondaryIcon, startDate } = course;
 
-  const label =
-    requiredDate < now ? 'upcoming' : hasDayInDate(course.startDate) ? 'available' : 'planned';
+  const now = dayjs();
+  const courseStartDate = dayjs(startDate);
 
-  const { title, altTitle, language, mode, enroll, secondaryIcon, startDate } = course;
+  const isAvailable =
+    hasDayInDate(startDate) &&
+    courseStartDate.isBetween(now.subtract(2, 'week'), now.add(2, 'week'));
+  const isUpcoming = courseStartDate.isBetween(now, now.add(3, 'month'));
+
+  let label = Labels.PLANNED;
+
+  if (isAvailable) {
+    label = Labels.AVAILABLE;
+  }
+
+  if (isUpcoming) {
+    label = Labels.UPCOMING;
+  }
```

Comment:
I'd extract it into a function outside the component

Reply:
Done